### PR TITLE
Metadata explorer click

### PR DESCRIPTION
**Pull request by IEvangelist**
Added the ability for the entries in the Metadata tree view to be clickable:

- Deleted "implicit" updates
- Major clean up
- Encapsulated a bunch of bits
- Make the traverse option a fallback if glob doesn't work
- Delete *\*-refresh.svg* images, as we can use codeicon instead

/cc: @gewarren

### CODE REVIEW

File: packages/docs-markdown/src/extension.ts [hunk 1]
```diff
@@ -201,8 +201,10 @@ export async function activate(context: ExtensionContext) {
 			}
 		}
 	});
-	workspace.onDidCloseTextDocument(e => {
-		metadataTreeProvider.refresh();
+	workspace.onDidChangeTextDocument(e => {
+		if (e.document.languageId === 'markdown') {
```
Comment (gewarren) [inline packages/docs-markdown/src/extension.ts:205]: Why did you change this? We want it to refresh and show the welcome text if the user closes the last Markdown file. Also if the user switches active focus to a non-Markdown file, we also want to refresh so we clear the tree (since it doesn't apply to the current editor).

File: packages/docs-markdown/src/controllers/metadata/metadata-tree-node.ts [hunk 2]
```diff
@@ -26,8 +36,27 @@ export class MetadataTreeNode extends TreeItem {
 		this.key = entry.key;
 		this.source = entry.source;
 		this.value = entry.value;
+		// this.resourceUri = entry.resourceUri;
```
[the remark this answers is not in the document]
Comment (IEvangelist) [reply packages/docs-markdown/src/controllers/metadata/metadata-tree-node.ts:39]:
```suggestion

```

File: packages/docs-markdown/src/extension.ts [hunk 3]
```diff
@@ -201,8 +201,10 @@ export async function activate(context: ExtensionContext) {
...
+			metadataTreeProvider.refresh();
+		}
```
Comment (IEvangelist) [inline packages/docs-markdown/src/extension.ts:207]:
```suggestion
		metadataTreeProvider.refresh();
```

File: packages/docs-markdown/src/extension.ts (hunk 1 above)
Comment (IEvangelist) [reply packages/docs-markdown/src/extension.ts:205]: I believe that this still works the way we want it, I do see the welcome page. I'll remove the guard for `'markdown'` though.

**Comment by IEvangelist** [comment]
Hi again @bharney0 - I'm fixing the bug in the unit test, as discovered from this test run failure. Should have it soon.

**Review by bharney0 — APPROVED** [review]
